docs/source/teams/data_lens.rst

```diff
@@ -0,0 +1,741 @@
+:class:`DataLensSearchResponse <fiftyone.operators.data_lens.models.DataLensSearchResponse>`
+objects. To start, we'll create some synthetic data to better understand the
+interaction between Data Lens and our operator. We'll look at a
+:ref:`more realistic example <data-lens-realistic-example>` later on.
+
+.. note::
+
+    Why a generator? Generators provide a convenient approach for long-lived,
+    lazy-fetching connections that are common in databases and data lakes.
+    While Data Lens does support operators which do not execute as generators,
+    we recommend using a generator during the beta period of this feature.
+
+.. _data-lens-generating-responses:
+
+Generating search responses
+---------------------------
+
+To adhere to the Data Lens interface, we need to yield
+:class:`DataLensSearchResponse <fiftyone.operators.data_lens.models.DataLensSearchResponse>`
+objects from our operator. A
+:class:`DataLensSearchResponse <fiftyone.operators.data_lens.models.DataLensSearchResponse>`
+is comprised of the following fields:
+
+-   `response.result_count`: a number indicating the number of samples being
+    returned in this response.
+-   `response.query_result`: a list of dicts containing serialized
+    :class:`Sample <fiftyone.core.sample.Sample>` data, e.g. obtained via
+    :meth:`to_dict() <fiftyone.core.sample.Sample.to_dict>`.
+
+.. note::
+
+    Data Lens expects sample data to adhere to the
+    :class:`fo.Sample <fiftyone.core.sample.Sample>` format. It is highly recommended that
+    you use the FiftyOne SDK to create your sample data.
+
+To see how Data Lens works, let's yield a response with a single synthetic
+sample.
+
+.. code-block:: python
+    :linenos:
+
+    def handle_lens_search_request(
+        self,
+        request: DataLensSearchRequest,
+        ctx: foo.ExecutionContext
+    ) -> Generator[DataLensSearchResponse, None, None]
+        # We'll use a placeholder image for our synthetic data
+        image_url = "https://placehold.co/150x150"
+
+        # Create a sample using the SDK
+        synthetic_sample = fo.Sample(filepath=image_url)
+
+        # Convert our samples to dicts
+        samples = [synthetic_sample.to_dict()]
+
+        # We'll ignore any inputs for now and yield a single response
+        yield DataLensSearchResponse(
+            result_count=len(samples),
+            query_result=samples
+        )
+
+Let's see what this looks like in Data Lens. After adding the operator as a
+data source, we can navigate to the "Query data" tab to interact with the
+operator. When we click the preview button, the Data Lens framework invokes
+our operator to retrieve sample data. Our operator yields a single sample, and
+we see that sample shown in the preview.
+
+.. image:: /images/teams/data_lens_synthetic_sample.png
+    :alt: data-lens-synthetic-sample
+    :align: center
+
+Let's modify our operator to incorporate the `request.batch_size` property.
```

Comment:
Any time you are attempting to have code, you technically need to use double backticks. In this case it is just a broken link that happens to look the way you wanted it to. I know, RST is dumb.

```
``request.batch_size``
```

Reply:
At the top of this file we're using 
```
.. default-role:: code
```
which I think takes care of this (i.e. these aren't being interpreted as refs afaik)

Reply:
Oh so that's what that does lol